File: benchmarks/reference/matrix_mul.java
public class matrix_mul {
    public static void main(String[] args) {
        long start = System.nanoTime();
        int size = 50;
        int[][] a = new int[size][size];
        int[][] b = new int[size][size];
        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                a[i][j] = i + j;
                b[i][j] = i - j;
            }
        }
        
        int[][] c = new int[size][size];
        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                int sumVal = 0;
                for (int k = 0; k < size; k++) {
                    sumVal += a[i][k] * b[k][j];
                }
                c[i][j] = sumVal;
            }
        }
        
        System.out.println("Matrix[0][0]: " + c[0][0]);
        double elapsed = (System.nanoTime() - start) / 1e9;
        System.out.println("Time: " + elapsed);
    }
}

File: benchmarks/reference/matrix_mul.cpp
#include <iostream>
#include <vector>
#include <chrono>

int main() {
    auto start = std::chrono::high_resolution_clock::now();
    int size = 50;
    std::vector<std::vector<int>> a(size, std::vector<int>(size));
    std::vector<std::vector<int>> b(size, std::vector<int>(size));
    
    for(int i=0; i<size; i++) {
        for(int j=0; j<size; j++) {
            a[i][j] = i + j;
            b[i][j] = i - j;
        }
    }
    
    std::vector<std::vector<int>> c(size, std::vector<int>(size));
    for(int i=0; i<size; i++) {
        for(int j=0; j<size; j++) {
            int sum_val = 0;
            for(int k=0; k<size; k++) {
                sum_val += a[i][k] * b[k][j];
            }
            c[i][j] = sum_val;
        }
    }
    
    std::cout << "Matrix[0][0]: " << c[0][0] << "\n";
    std::chrono::duration<double> elapsed = std::chrono::high_resolution_clock::now() - start;
    std::cout << "Time: " << elapsed.count() << "\n";
    return 0;
}
